transaction/undoTx.go

```diff
@@ -130,19 +133,41 @@ func NewUndoTx() TX {
 }
 
 func releaseUndoTx(t *undoTx) {
-	t.abort()
+	// Reset the pointers in the log entries, but need not allocate a new
+	// backing array
+	t.abort(false)
 	undoPool <- t
 }
 
-func (t *undoTx) resetLogTail() {
+func (t *undoTx) setTail(tail int) {
+	t.tail = tail
+	runtime.PersistRange(unsafe.Pointer(&t.tail), unsafe.Sizeof(t.tail))
+}
+
+// The realloc parameter indicates if the backing array for the log entries
+// need to be reallocated.
+func (t *undoTx) resetLogTail(realloc bool) {
+	tail := t.tail
 	runtime.Fence()
 	if t.tail != 0 {
-		t.tail = 0
-		runtime.PersistRange(unsafe.Pointer(&t.tail), unsafe.Sizeof(t.tail))
+		t.setTail(0)
+	}
+
+	if realloc || cap(t.log) > NUMENTRIES {
+		// Allocate a new backing array if realloc is true or if the array was
```

Comment:
We could do that. We may then have to consider if zeroing out the entries will become costlier than allocating a new array. Also, keeping around a large log array consumes that much memory when it might not be required?
We could identify a log length threshold at which we do one or the other approach. But I didn't want to complicate this piece of code.

Reply:
Reusing the log buffers when cap(t.log) > NUMENTRIES turned out to be faster than allocating a new array. So I'll change that line of code in a new pull request.